packages/core/src/RenderingEngine/Viewport.ts

```diff
@@ -1437,6 +1577,58 @@ class Viewport implements IViewport {
     return true;
   }
 
+  /**
+   * Gets a basic view reference.
+   */
+  public getViewPresentation(
+    viewPresSel: ViewPresentationSelector = {
+      rotation: true,
+      displayArea: true,
+      zoom: true,
+      pan: true,
+    }
+  ): ViewPresentation {
+    const target: ViewPresentation = {};
+
+    const { rotation, displayArea, zoom, pan } = viewPresSel;
+    if (rotation) {
+      target.rotation = this.getRotation();
+    }
+    if (displayArea) {
+      target.displayArea = this.getDisplayArea();
+    }
+    const initZoom = this.getZoom();
+
+    if (zoom) {
+      target.zoom = initZoom;
+    }
+    if (pan) {
+      target.pan = this.getPan();
+      vec2.scale(target.pan, target.pan, 1 / initZoom);
+    }
+    return target;
```

Comment:
i'm not sure why we don't return all of them? and at the destination we use the selector too choose speicifc one? aren't these pure functoins? they are all get... (getRotation, getDisplayArea) etc